src/tools/suggest-tests/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use std::{
+    fmt::{self, Display},
+    path::Path,
+};
+
+use glob::Pattern;
+
+use build_helper::git::get_modified_files_since_diverge;
+
+use crate::{dynamic_suggestions::DYNAMIC_SUGGESTIONS, static_suggestions::STATIC_SUGGESTIONS};
+
+mod dynamic_suggestions;
+mod static_suggestions;
+
+pub fn get_suggestions(verbose: bool) -> Result<Vec<Suggestion>, String> {
+    let modified_files = get_modified_files_since_diverge(None);
+    let modified_files = match modified_files {
+        Ok(files) => files,
+        Err(err) => return Err(format!("Could not get modified files from git: \"{err}\"")),
+    };
+
+    if verbose {
+        println!("=== MODIFIED FILES ===");
+        for file in &modified_files {
+            println!("{}", file);
+        }
+        println!("======================\n");
+    }
+
+    let mut suggestions = Vec::new();
+
+    // static suggestions
+    for sug in STATIC_SUGGESTIONS.iter() {
+        let glob = Pattern::new(&sug.0).expect("Found invalid glob pattern!");
+
+        for file in &modified_files {
+            if glob.matches(file) {
+                suggestions.extend_from_slice(&sug.1);
+            }
+        }
+    }
+
+    // dynamic suggestions
+    for sug in DYNAMIC_SUGGESTIONS {
+        for file in &modified_files {
+            let sugs = sug(Path::new(file));
+
+            suggestions.extend_from_slice(&sugs);
+        }
+    }
+
+    suggestions.sort();
+    suggestions.dedup();
+
+    println!("=== SUGGESTIONS ===");
+    for sug in &suggestions {
+        println!("{}", sug);
+    }
+    println!("===================");
```

Comment:
The output here is not great if there are 0 suggestions: maybe we can print something like "No suggestions found" and then either "Consider adding a new rule" or "Modify at least 1 file to get suggestions"?
```
=== SUGGESTIONS ===
===================
help: consider using the `--run` flag to automatically run suggested tests
Build completed successfully in 0:00:13
```